### PR TITLE
Swap out Unirest for OkHttp3

### PR DESCRIPTION
**Pull request by rubikzube**
- [x] Replaced Unirest with OkHttp which is better maintained and has fewer shutdown issues

### CODE REVIEW

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 1]
```diff
@@ -50,32 +57,7 @@ public ConstructorIO(String apiToken, String apiKey, boolean isHTTPS, String hos
...
-            if (!isFirst) {
-                urlString += "&";
-            }
-            urlString += URLEncoder.encode(key, "UTF-8");
-            urlString += "=";
-            urlString += URLEncoder.encode(params.get(key), "UTF-8");
-            isFirst = false;
-        }
-        return urlString;
+        this.credentials = "Basic " + Base64.getEncoder().encodeToString((this.apiToken + ":").getBytes());
```
Comment (Shmaltalk) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java]: Out of curiosity, why encode to base64?

Comment (Shmaltalk) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java]: Oh also why add the colon?

Comment (rubikzube) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java]: https://tools.ietf.org/html/rfc2617#section-2

Comment (Shmaltalk) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java]: oh.

**Pull request merged**